### DISCUSSION TITLE
Conditional page runtime fallback to nodejs

### DISCUSSION THREAD

**Discussion by wenerme** [vercel/next.js#47906] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 8.1.0
    Relevant packages:
      next: 13.2.4
      eslint-config-next: 13.2.3
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Standalone mode (output: "standalone")

### Link to the code that reproduces this issue

https://github.com/wenerme/ChatGPT-Next-Web/blob/b0907ac44fc623b79348bd3fcccff1ad48d712f6/app/api/chat-stream/route.ts#L60-L62

### To Reproduce

```ts
export const config: PageConfig = {
  runtime: process.env.NEXT_RUNTIME,
};
```

> Unsupported node type "MemberExpression"

Or

```ts
export const config: PageConfig = {
  runtime: process.env.STANDALONE ? "nodejs" : "edge",
};
```

> Unsupported node type "ConditionalExpression"

### Describe the Bug

PageConfig should allowed simple expression

### Expected Behavior

Conditional page runtime works as expected

**Comment by king8fisher** [comment]
I Second this.
With 'experimental-edge', local `dev` mode fails with 'Internal Server Error'. Without the setting, cloudflare complaints and fails to build. 
Is there a way to rewrite the setting dynamically?
